crypto/dilithium/pqcrystals_dilithium_ref_common/params.c

```diff
@@ -0,0 +1,82 @@
+#include <openssl/base.h>
+#include <assert.h>
+
+#include "params.h"
+
+static void ml_dsa_params_init(ml_dsa_params *params, size_t k) {
+  assert((k == 2) || (k == 3) || (k == 5));
+  size_t K;
+  size_t L;
+  size_t tau;
+  size_t beta;
+  size_t omega;
+  size_t c_tilde_bytes;
+
+  if (k == 2) {
+    K = 4;
+    L = 4;
+    tau = 39;
+    beta = 78;
+    omega = 80;
+    c_tilde_bytes = 32;
+  }
+  else if (k == 3) {
+    K = 6;
+    L = 5;
+    tau = 49;
+    beta = 196;
+    omega = 55;
+    c_tilde_bytes = 48;
+  }
+  else { //k == 5
+    K = 8;
+    L = 7;
+    tau = 60;
+    beta = 120;
+    omega = 75;
+    c_tilde_bytes = 64;
+  }
+
+  size_t eta = (k == 3) ? 4 : 2;
```

Comment:
While this helps avoids some duplicate values (e.g. both K = 2 and K = 5 get eta = 2), this would be easier to read and cross reference with the deleted values in params.h if they were all just defined up in the big if chain.

Reply:
I agree with you on this. More clarity in [c37f75b](https://github.com/aws/aws-lc/pull/1910/commits/c37f75baaf20d6724cb877b2360c7a5e104c8bb9).